requests_toolbelt/utils/dump.py

```diff
@@ -115,12 +219,17 @@ def _coerce_to_bytes(data):
 
 
 def dump_response(response, request_prefix=b'< ', response_prefix=b'> ',
-                  data_array=None):
+                  data_array=None, sanitizer=NOOP_SANITIZER):
```

Comment:
Let's leave this as having a default of `None` and then instantiate a new `NoopSanitizer` if it is None. (Same for above.)

Reply:
Why? I don't see the upside to be honest, so it would be appreciated if you could enlighten me.

Downsides of doing as you suggest (as I see it):
- Hide the actual behavior from the signature, since you no longer see that it uses `NoopSanitizer`
- Complicate (slightly) the body of the function, since we now have to do a `None`-check and instantiate a new `NoopSanitizer`
- Performance-drop if we instantiate a `NoopSantiizer` on every call

Reply:
> Hide the actual behavior from the signature, since you no longer see that it uses NoopSanitizer

The behaviour should be documented in the doc-string regardless. I'm not sure the behaviour belongs in the signature at all.

> Complicate (slightly) the body of the function, since we now have to do a None-check and instantiate a new NoopSanitizer

I don't believe that `sanitizer = sanitizer or NOOP_SANITIZER` or even `sanitizer = sanitizer or NoopSanitizer()` is that big of a complication.

> Performance-drop if we instantiate a NoopSantiizer on every call

```
❯❯❯ python -m timeit -s 'from requests_toolbelt.utils.dump import Sanitizer' -- 'Sanitizer()'    ⏎ feature/redact-sensitive-headers
10000000 loops, best of 3: 0.108 usec per loop
```

And that's on a sad little old machine. Anything remotely modern or performant should have little trouble with the added instantiation.

Further, comparing it to the rest of the `dump_*` functions which iterate over potentially large dictionaries, it really can't be such a significant hit as to prefer the minor potential performance improvement over correctness.

Reply:
I still don't see the upside to be honest, but as you say, most of my objections are really minor points. I'll change it.